### PR TITLE
[stable/jenkins] Support servicemonitor and alerting rules

### PR DESCRIPTION
**Pull request by startnow65**
#### What this PR does / why we need it:

Adds support for adding Prometheus service monitor and alerting rules

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/jenkins/templates/jenkins-master-alerting-rules.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+---
+apiVersion: monitoring.coreos.com/v1
+kind: PrometheusRule
+metadata:
+  name: {{ template "jenkins.fullname" . }}
+  labels:
+    "app.kubernetes.io/name": '{{ template "jenkins.name" .}}'
+    "helm.sh/chart": "{{ .Chart.Name }}-{{ .Chart.Version }}"
+    "app.kubernetes.io/managed-by": "{{ .Release.Service }}"
+    "app.kubernetes.io/instance": "{{ .Release.Name }}"
```
Comment (torstenwalter) [inline stable/jenkins/templates/jenkins-master-alerting-rules.yaml:11]: I would add the label `"app.kubernetes.io/component": "{{ .Values.master.componentName }}"`

Comment (startnow65) [reply stable/jenkins/templates/jenkins-master-alerting-rules.yaml:11]: @torstenwalter added, thanks

**Comment by torstenwalter** [comment]
@startnow65 Than you for this PR. It's a really great addition to have the option to configure monitoring and alerting.
Next steps could be a default grafana dashboard or examples for alerting rules.

**Comment by startnow65** [comment]
> @startnow65 Than you for this PR. It's a really great addition to have the option to configure monitoring and alerting.
> Next steps could be a default grafana dashboard or examples for alerting rules.

Thanks @torstenwalter, I have some alerting rules defined for my setup. Would monitor for a while and if it makes sense would submit a PR with these rules and respective grafana dashboards

**Pull request merged**